components/Places/ImagePicker.js

```diff
@@ -0,0 +1,79 @@
+import { View, Button, Text, Alert, Image, StyleSheet } from "react-native";
+import {
+  launchCameraAsync,
+  useCameraPermissions,
+  PermissionStatus,
+} from "expo-image-picker";
+import { useState } from "react";
+import { Colors } from "../../constants/colors";
+import OutlinedButton from "../UI/OutlinedButton";
+
+function ImagePicker() {
+  const [pickedImage, setPickedImage] = useState();
+
+  const [cameraPermissionsInformation, requestPermission] =
+    useCameraPermissions();
+  async function verifyPermission() {
+    if (cameraPermissionsInformation.status === PermissionStatus.UNDETERMINED) {
+      // we don't know yet if we have access or not
+      const permissionResponse = await requestPermission();
+
+      return permissionResponse.granted;
+    }
+    if (cameraPermissionsInformation.status === PermissionStatus.DENIED) {
+      Alert.alert(
+        "Permission Denied!",
+        "You need to give camera permission to use this app."
+      );
+      return false;
+    }
+    return true;
+  }
+  async function takeImageHandler() {
+    const hasPermission = await verifyPermission();
+
+    if (!hasPermission) {
+      return;
+    }
+
+    const image = await launchCameraAsync({
+      allowsEditing: true,
+      aspect: [16, 9],
+      quality: 0.5,
+    });
+    setPickedImage(image.uri);
+  }
+
+  let imagePreview = <Text>No image taken yet.</Text>;
+
+  if (pickedImage) {
+    imagePreview = <Image style={styles.image} source={{ uri: pickedImage }} />;
+  }
+
+  return (
+    <View>
+      <View style={styles.imagePreview}>{imagePreview}</View>
```

Comment:
imagePreview'i bir değişken gibi tutmak yerine bir fonksiyon yazsak ve bize gereken ekranı döndürse daha okunur ve temiz bir kod olur. imagePreview'in son halini takip etmek için kodda arama yapmam gerekti bunun yerine fonksiyon olsa bakacağım yeri daha net şekilde bulurum.

```
const renderImagePreview = () => {
if (pickedImage) {
return <Image style={styles.image} source={{ uri: pickedImage }} />;
}

return <Text>No image taken yet.</Text>
}
```

ve bu fonksiyonu kullanacağımız yerde çağıracağız.
`<View style={styles.imagePreview}>{renderImagePreview()}</View>`